test_scripts/API/VehicleData/OnVehicleData/001_OnVD_Success.lua

```diff
@@ -16,22 +16,21 @@
 --[[ Required Shared libraries ]]
 local common = require('test_scripts/API/VehicleData/common')
 
+--[[ Local Constants ]]
+local testTypes = {
+  common.testType.VALID_RANDOM_ALL,
+  common.testType.VALID_RANDOM
+}
+
 --[[ Scenario ]]
 common.Title("Preconditions")
 common.Step("Clean environment and update preloaded_pt file", common.preconditions)
 common.Step("Start SDL, HMI, connect Mobile, start Session", common.start)
 common.Step("Register App", common.registerApp)
+common.Step("Activate App", common.activateApp)
 
 common.Title("Test")
-for param in common.spairs(common.getVDParams(true)) do
-  common.Title("VD parameter: " .. param)
-  common.Step("RPC " .. common.rpc.sub, common.processSubscriptionRPC, { common.rpc.sub, param })
-  common.Step("RPC " .. common.rpc.on, common.sendOnVehicleData, { param, common.isExpected })
-end
-for param in common.spairs(common.getVDParams(false)) do
-  common.Title("VD parameter: " .. param)
-  common.Step("RPC " .. common.rpc.on, common.sendOnVehicleData, { param, common.isNotExpected })
-end
+common.getTestsForOnVD(testTypes)
```

Comment:
Duplicated checks for parent parameters can be removed.

Reply:
Done in b3d3d3b